<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ONetworkDebuger</class>
 <widget class="QMainWindow" name="ONetworkDebuger">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>664</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_5">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_11">
      <item>
       <widget class="QTextEdit" name="Output">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>20</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>信息输出区，收到的包和提示信息将会在这里显示</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="html">
         <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Ubuntu'; font-size:11pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;界面有限，鼠标悬停在各控件上，可获得更多提示...&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_10">
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>监听/连接列表</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="DoClearOutput">
            <property name="maximumSize">
             <size>
              <width>30</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string>清空左面输出区域的文字</string>
            </property>
            <property name="text">
             <string>清屏</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="DoInitialize">
            <property name="maximumSize">
             <size>
              <width>50</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string>停止所有监听、断开所有连接</string>
            </property>
            <property name="text">
             <string>初始化</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <item>
           <widget class="QCheckBox" name="ShowSendMsg">
            <property name="toolTip">
             <string>勾选此复选框，将会在左面的输出区显示已经发出的包的信息</string>
            </property>
            <property name="text">
             <string>显示发出的包</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QListWidget" name="ConnectList">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>20</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>190</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="toolTip">
           <string>监听的端口、已经建立的连接将在这里显示。双击其中的条目会停止监听或断开连接</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>监听UDP端口</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QSpinBox" name="UdpListenPort">
           <property name="minimumSize">
            <size>
             <width>70</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>70</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="maximum">
            <number>65535</number>
           </property>
           <property name="value">
            <number>4320</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="DoUdpListen">
           <property name="minimumSize">
            <size>
             <width>55</width>
             <height>30</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>55</width>
             <height>30</height>
            </size>
           </property>
           <property name="text">
            <string>监听</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>建立TCP连接</string>
        </property>
        <property name="flat">
         <bool>false</bool>
        </property>
        <property name="checkable">
         <bool>false</bool>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLineEdit" name="TcpConnectIp">
           <property name="text">
            <string>127.0.0.1</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="TcpConnectPort">
           <property name="minimumSize">
            <size>
             <width>70</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>70</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="maximum">
            <number>65535</number>
           </property>
           <property name="singleStep">
            <number>1</number>
           </property>
           <property name="value">
            <number>4321</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="DoTcpConnect">
           <property name="minimumSize">
            <size>
             <width>55</width>
             <height>30</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>55</width>
             <height>30</height>
            </size>
           </property>
           <property name="text">
            <string>连接</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="title">
         <string>监听TCP端口</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QSpinBox" name="TcpListenPort">
             <property name="minimumSize">
              <size>
               <width>70</width>
               <height>0</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>70</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="maximum">
              <number>65535</number>
             </property>
             <property name="value">
              <number>4321</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="DoTcpListen">
             <property name="minimumSize">
              <size>
               <width>55</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>55</width>
               <height>30</height>
              </size>
             </property>
             <property name="text">
              <string>监听</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="AutoAccept">
             <property name="maximumSize">
              <size>
               <width>35</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="toolTip">
              <string>按下该按钮，将自动接受新的连接，否则会在有新连接时提示</string>
             </property>
             <property name="text">
              <string>Auto</string>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_9">
      <item>
       <widget class="QGroupBox" name="groupBox_4">
        <property name="title">
         <string>发送UDP数据包</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <property name="sizeConstraint">
            <enum>QLayout::SetMaximumSize</enum>
           </property>
           <item>
            <widget class="QComboBox" name="UdpSendSelect">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="toolTip">
              <string>在这里选择或输入发送端口，也可以选择随机端口</string>
             </property>
             <property name="editable">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="UdpSendIp">
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>27</height>
              </size>
             </property>
             <property name="text">
              <string>127.0.0.1</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="UdpSendPort">
             <property name="minimumSize">
              <size>
               <width>70</width>
               <height>0</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>70</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="maximum">
              <number>65535</number>
             </property>
             <property name="value">
              <number>4320</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QGridLayout" name="gridLayout">
           <property name="sizeConstraint">
            <enum>QLayout::SetMaximumSize</enum>
           </property>
           <item row="0" column="0">
            <widget class="QSpinBox" name="UdpSendHeadVersion">
             <property name="toolTip">
              <string>协议版本号，默认为2</string>
             </property>
             <property name="minimum">
              <number>-2147483647</number>
             </property>
             <property name="maximum">
              <number>2147483647</number>
             </property>
             <property name="value">
              <number>2</number>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QSpinBox" name="UdpSendHeadLength">
             <property name="toolTip">
              <string>消息数据部分长度(不含消息头),您可以勾选下面的&quot;自动计算长度&quot;复选框，使其自动填写</string>
             </property>
             <property name="maximum">
              <number>999999999</number>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QSpinBox" name="UdpSendHeadMsgType">
             <property name="toolTip">
              <string/>
             </property>
             <property name="minimum">
              <number>-2147483647</number>
             </property>
             <property name="maximum">
              <number>2147483647</number>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLineEdit" name="UdpSendHeadTime">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="autoFillBackground">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>0</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QTextEdit" name="UdpSendContent">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Ignored">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>60</height>
            </size>
           </property>
           <property name="sizeIncrement">
            <size>
             <width>0</width>
             <height>60</height>
            </size>
           </property>
           <property name="baseSize">
            <size>
             <width>0</width>
             <height>60</height>
            </size>
           </property>
           <property name="toolTip">
            <string>消息的数据部分(字符串)</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <property name="sizeConstraint">
            <enum>QLayout::SetMaximumSize</enum>
           </property>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QCheckBox" name="UdpSendAutoCalcLength">
             <property name="toolTip">
              <string>勾选此复选框，将会自动在每次发送消息前计算消息数据部分长度</string>
             </property>
             <property name="text">
              <string>自动长度</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="UdpSendAutoFillCurrentTime">
             <property name="toolTip">
              <string>勾选此复选框，将会自动在每次发送消息前填入当前时间</string>
             </property>
             <property name="text">
              <string>自动时间</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="UdpDoSend">
             <property name="minimumSize">
              <size>
               <width>55</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>55</width>
               <height>30</height>
              </size>
             </property>
             <property name="text">
              <string>发送</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_5">
        <property name="title">
         <string>发送TCP数据包</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QComboBox" name="TcpSendSelect">
           <property name="toolTip">
            <string>请选择TCP连接对象以发送数据包</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="0">
            <widget class="QSpinBox" name="TcpSendHeadVersion">
             <property name="toolTip">
              <string>协议版本号，默认为2</string>
             </property>
             <property name="minimum">
              <number>-2147483647</number>
             </property>
             <property name="maximum">
              <number>2147483647</number>
             </property>
             <property name="singleStep">
              <number>1</number>
             </property>
             <property name="value">
              <number>2</number>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QSpinBox" name="TcpSendHeadLength">
             <property name="toolTip">
              <string>消息数据部分长度(不含消息头),您可以勾选下面的&quot;自动计算长度&quot;复选框，使其自动填写</string>
             </property>
             <property name="maximum">
              <number>999999999</number>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QSpinBox" name="TcpSendHeadMsgType">
             <property name="minimum">
              <number>-2147483647</number>
             </property>
             <property name="maximum">
              <number>2147483647</number>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="TcpSendHeadTime">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>0</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QTextEdit" name="TcpSendContent">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Ignored">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>60</height>
            </size>
           </property>
           <property name="sizeIncrement">
            <size>
             <width>0</width>
             <height>60</height>
            </size>
           </property>
           <property name="baseSize">
            <size>
             <width>0</width>
             <height>60</height>
            </size>
           </property>
           <property name="toolTip">
            <string>消息的数据部分(字符串)</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_7">
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QCheckBox" name="TcpSendAutoCalcLength">
             <property name="toolTip">
              <string>勾选此复选框，将会自动在每次发送消息前计算消息数据部分长度</string>
             </property>
             <property name="text">
              <string>自动长度</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="TcpSendAutoFillCurrentTime">
             <property name="toolTip">
              <string>勾选此复选框，将会自动在每次发送消息前填入当前时间</string>
             </property>
             <property name="text">
              <string>自动时间</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="TcpDoSend">
             <property name="minimumSize">
              <size>
               <width>55</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>55</width>
               <height>30</height>
              </size>
             </property>
             <property name="text">
              <string>发送</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>664</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>功能(服务器)</string>
    </property>
    <addaction name="action"/>
    <addaction name="action_SHA_1"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>测试(服务器)</string>
    </property>
    <addaction name="action_10W"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action">
   <property name="text">
    <string>将密码加密为登录字符串(用户名/密码/公钥)</string>
   </property>
  </action>
  <action name="action_SHA_1">
   <property name="text">
    <string>将密码加密为SHA-1串(用户名/密码)</string>
   </property>
  </action>
  <action name="action_10W">
   <property name="text">
    <string>连接/断开10W次</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
